README.md

```diff
@@ -4,4 +4,149 @@
 + Explain what hoisting is
 + Explain why hoisting is important to remember
 + Avoid variable hoisting
-+ Avoid function hoisting
\ No newline at end of file
++ Avoid function hoisting
+
+
+## What Is Hoisting
+
+Hoisting is when variables and/or functions are hoisted or lifted to the top of the scope in which they were defined. JavaScript was originally intended to be written in a single file. When you're working on a large code base in a single file, you can't always define variables in the order in which you need them, which is why hoisting can be helpful.
+
+Because we now write JavaScript in multiple files for large applications, hoisting can primarily get us into trouble.
+
+## Variable Hoisting
+
+While talking about hoisting, it's important to remember that JavaScript has function-level scope, which means that functions have access to variables that are defined outside of it, but any variables defined inside of a function are not accessible outside.
+
+This is very different from Ruby. Let's take this example in Ruby:
+
+```ruby
+def my_method
+  puts word
+  word = "cake"
+end
+```
+This code would blow up with the undefined method or variable error because we're using the variable `word` before it's defined.
+
+Let's take the same example in JavaScript:
+
+```js
+function myFunction(){
+  console.log(word)
+  var word = "cake";
+}
+```
+This JavaScript would print out `undefined`. That's because variable hoisting just raises the **variable declaration**. JavaScript reads the above code like this:
+
+```js
+function myFunction(){
+  var word;
+  console.log(word);
+  var word = "cake";
+}
+```
+
+We see `undefined` printed in the console because JavaScript reads the variable as declared, but not defined.
+
+Let's start with an example of JavaScript accessing a variable inside a function that was defined outside:
+
+```js
+var snack = "granola bar";
+function eat(){
+  console.log(snack);
+}
+eat();
+// prints out "granola bar" and returns undefined
+```
+
+Let's modify the function a little bit: 
+```js
+var snack = "granola bar";
+function eat(){
+  console.log(snack);
+  var snack = "snickers";
+}
+eat();
+// prints out undefined, returns undefined
+console.log(snack);
+// prints out "granola bar"
```

Comment:
Can we add a code sample here to demonstrate what JS does with the hoisting? I was still confused by this example:

```js
var snack = "granola bar";
function eat() {
  var snack;
  console.log(snack);
  snack = "snickers";
}

eat(); // prints undefined
console.log(snack); // prints granola bar